python_modules/dagster/dagster/components/core/load_defs.py

```diff
@@ -91,3 +94,33 @@ def load_defs(defs_root: ModuleType, project_root: Optional[Path] = None) -> Def
             root_component.build_defs(context),
             Definitions(metadata={PLUGIN_COMPONENT_TYPES_JSON_METADATA_KEY: components_json}),
         )
+
+
+TComponent = TypeVar("TComponent", bound=Component)
+
+
+@public
+@preview
+def get_components(
+    path: Path,
+    of_type: type[TComponent] = Component,
+) -> list[TComponent]:
+    """Get Component instances from the project for the given path,
+    optionally filtering via of_type.
+    """
+    # defer imports for optional deps
+    from dagster_dg.context import DgContext
+
+    components = []
+
+    dg_context = DgContext.for_project_environment(path, command_line_config={})
```

Comment:
im also not high confidence about using dg context instead of something else for this